Feature/1495 Add KEWR

CanisLupus518 opened the pull request:
Resolves #1495 

The purpose of this pull request is to add Newark Int’l Airport

CanisLupus518 commented:
@erikquinn @Fechulo I think this is pretty solid now. I have to work on the airport guide, but other than that, all it needs is terrain. I haven't been able to figure out the terrain generation.

Fechulo commented on assets/airports/kewr.json:
```diff
@@ -0,0 +1,5706 @@
+                "KEWR22L": [["_RWY22L01DME", "A05+"]],
+                "KEWR22R": [["_RWY22R01DME", "A05+"]]
+            },
+            "body": ["BAGGA", ["PORTT", "A25"]],
```
BAGGA should S220-

Fechulo commented on assets/airports/kewr.json:
```diff
@@ -0,0 +1,5706 @@
+                "BREZY": ["#261", "BREZY"],
+                "COATE": ["#261", "COATE"],
+                "DIXIE": ["#261", "ELVAE", "COL","DIXIE"],
+                "ELIOT": ["SIRLY", "FENLY", "ELIOT","ELIOT"],
```
ELIOT is written twice

Fechulo commented on assets/airports/kewr.json:
```diff
@@ -0,0 +1,5706 @@
+            "icao": "LIB5",
+            "name": "Liberty Five",
+            "rwy": {
+                "KEWR22L": [["_RWY22L01DME", "A05+"], ["#215", "A025"]],
```
As with the EWR4, I'm not sure what the purpose of the altitude restriction on a heading is

CanisLupus518 replied:
I wasn't sure how else to get the aircraft to stop their climb and wait for ATC. With the addition of the new commands feature in the spawnPatterns, this can probably go away and be handled in the spawn pattern.

Fechulo commented on assets/airports/kewr.json:
```diff
@@ -0,0 +1,5706 @@
+                "GVE": ["GVE", "OTT", "PALEO", "PEEDS", ["FUBRR", "A270+"]],
+                "FAK": ["FAK", "OTT", "PALEO", "PEEDS", ["FUBRR", "A270+"]],
+                "PXT": ["PXT", "PEEDS", ["FUBRR", "A270+"]],
+                "DQO": ["DQO"]
```
DQO should be in the body and A200+

Fechulo commented on assets/airports/kewr.json:
```diff
@@ -0,0 +1,5706 @@
+            "name": "Flossy Four",
+            "entryPoints": {
+                "GEE":   ["GEE", "EXTOL", "KODEY", "HNK", "HELON", "FLOSI"],
+                "BAF":   ["BAF", "BOWAN", "WIGAN", "TALCO", "WEETS", "FLOSI"],
```
missing HELON after WEETS

Fechulo commented on assets/airports/kewr.json:
```diff
@@ -0,0 +1,5706 @@
+            "entryPoints": {
+                "GEE":   ["GEE", "EXTOL", "KODEY", "HNK", "HELON", "FLOSI"],
+                "BAF":   ["BAF", "BOWAN", "WIGAN", "TALCO", "WEETS", "FLOSI"],
+                "HANAA": ["HANAA", "ALB", "ACOVE", "TALCO", "WEETS", "FLOSI"],
```
missing HELON after WEETS

Fechulo commented on assets/airports/kewr.json:
```diff
@@ -0,0 +1,5706 @@
+                "NELIE": ["NELIE", "IGN", "FLOSI"],
+                "FLOSI": ["FLOSI"]
+            },
+            "body": [["CRANK", "A70"], "SHAFF", "SAX", "PHLBO", "HOKIR", "#180"],
```
HOKIR is a fly-over waypoint denoted with a `^`

Fechulo commented on assets/airports/airportLoadList.json:
```diff
@@ -198,6 +198,12 @@
     },
+    {
+        "icao": "kewr",
+        "level": "easy",
```
With 69.625 acph, this falls under 'hard'

Fechulo reviewed (changes requested):
A few small changes, very fun to play

CanisLupus518 commented on assets/airports/kewr.json:
```diff
@@ -0,0 +1,5706 @@
+            "icao": "EWR4",
+            "name": "Newark Four",
+            "rwy": {
+                "KEWR4L" : [["_RWY04L01DME", "A05+"], "_RWY04L04DME", ["#290", "A030"]],
```
![image](https://user-images.githubusercontent.com/45713872/147397923-a05457f3-7ccc-4097-81be-ec049eededce.png)

CanisLupus518 replied:
The top altitudes are different for each runway, and generally assigned at clearance time. The text description in the chart mentions the altitudes used in the file. The aircraft appear to handle the altitude restriction on the heading segment just fine, but I am open to handling it differently if you have suggestions.

Fechulo replied:
Ahh, I see. If it ain't broke, don't fix it, I guess. I can't think of a better way of writing it so it can stay, I just don't like it on principle 😅

CanisLupus518 commented:
@Fechulo my last commit should address all of your requested changes except for the first. I may need more guidance or discussion there.

Fechulo commented:
Yeah, as far as I know, there is no way to define separate top altitudes for each SID, be it in the SID definition or spawnpattern. (@erikquinn correct me if I'm wrong, I know some stuff has changed and I haven't been very active lately) 

So although I don't like it, if placing an altitude restriction on a heading works, then I guess we can leave it in

CanisLupus518 commented:
@Fechulo I moved the heading altitudes into the spawn pattern using  newer feature that allows commands to be given to aircraft as they are spawned. I think maybe this addresses your concerns?

Fechulo commented:
That's a lot nicer, brilliant!

Could I ask you to add a few ILS fixes? Having the IAF and FAF on screen is super useful for beginners

CanisLupus518 commented:
> That's a lot nicer, brilliant!
> 
> Could I ask you to add a few ILS fixes? Having the IAF and FAF on screen is super useful for beginners

Will do

CanisLupus518 commented:
@Fechulo fixes added

CanisLupus518 commented:
@Fechulo given the revelation that I can snap vertices together in qGIS (I didn't know that until today) and the discussion about the first airspace poly not being super important. I would like to make one last commit here with nothing but an airspace change. It removes the first poly. I also went through every poly and snapped the vertices together between them to ensure there are no gaps in the airspace. I know you approved the PR already, but would it be ok to add?

Fechulo commented:
@CanisLupus518 go for it!

CanisLupus518 commented:
> @CanisLupus518 go for it!

done

CanisLupus518 commented:
@erikquinn @Fechulo I found a couple of syntax issues in the EWR file after playing around with it. I know this PR is closed, but the feature has yet to be released... what is the proper way of making a quick minor change? Creating a new issue? reopening the PR?
